### ISSUE TITLE
Cannot read property 'ownerInjection' of undefined

### ISSUE THREAD

**Issue by csetera** [simplabs/ember-simple-auth#1587]
I'm not sure if this is related to Issue #1490 or is something separate.  We are in the midst of trying to upgrade our application to a newer version of Ember (2.18 LTS in this case) and also picking up most recent versions of our various dependencies.  However, we are seeing the following stack trace during startup in our application as well as a fresh Ember app that only has ember-simple-auth.  We have also tried taking the application to Ember 3.x with the same results.

`adaptive.js:140 Uncaught TypeError: Cannot read property 'ownerInjection' of undefined
    at Class._createStore (adaptive.js:140)
    at Class.init (adaptive.js:125)
    at Class.superWrapper [as init] (ember-utils.js:418)
    at new Class (core_object.js:113)
    at Function._ClassMixinProps.create (core_object.js:282)
    at Class.init (application.js:20)
    at Class.superWrapper [as init] (ember-utils.js:418)
    at new Class (core_object.js:113)
    at Function._ClassMixinProps.create (core_object.js:282)
    at FactoryManager.create (container.js:439)`

At this point, we are stumped on how to proceed.  We depend on ember-simple-auth for our application and at the moment appear to be stuck.  Any suggestions would be appreciated.

**Comment by csetera** [comment]
A bit more information to add to the ticket.  I started playing around with different versions of ember-simple-auth.  I'm finding that if I lock the version to 1.4.0, then I don't see this issue.  Anything greater than 1.4.0 causes this issue to occur.  I'm attaching the simple test project I was using to recreate the issue as well as a listing of the dependencies for working and not working.  Let me know if there is anything else I can do to help narrow this down.  For now, I'm going to pin our version to 1.4.0 and see if I can make further progress with our updates.

[failing-1.6.0.txt](https://github.com/simplabs/ember-simple-auth/files/1919740/failing-1.6.0.txt)
[simple-auth-test.zip](https://github.com/simplabs/ember-simple-auth/files/1919741/simple-auth-test.zip)
[working-1.4.0.txt](https://github.com/simplabs/ember-simple-auth/files/1919742/working-1.4.0.txt)

**Comment by marcoow** [comment]
It might still be easier to extend than to wrap even if you're actually doing real work in addition to what the `AdaptiveStore` does.

**Comment by csetera** [comment]
Agreed.  I can't remember why I did it that way.... almost 3 years ago, so my memory of reasoning is a bit foggy :-)

**Comment by csetera** [comment]
Quick update.. extending the Adaptive store fixed this issue.  Thanks so much for the help!
